### PR TITLE
Improve feed rendering.

### PR DESCRIPTION
**Pull request by matthewwardrop**
This patch set improves the rendering of the knowledge post index feed by:
 - Making rows more compact and with more consistent heights (descriptions are collapsed)
 - Visual feedback for interaction (backgrounds on hover, depression when clicked)
 - Making the who row clickable to take you to the post proper

The new javascript methods are encapsulated in a closure so that they do not clutter the global namespace, and both the javascript and css files are loaded only on the page they are required.

Below are some screenshots of the new theming:

screen shot 2017-02-08 at 11 44 19 am
screen shot 2017-02-08 at 12 24 33 pm

Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj

### CODE REVIEW

File: knowledge_repo/app/static/css/custom.css [hunk 1]
```diff
@@ -256,29 +241,6 @@ a.label:focus, a.label:hover {
   color: #9CA299;
 }
 
-.feed-author {
-  padding-bottom: 5px;
```
Comment (NiharikaRay) [inline knowledge_repo/app/static/css/custom.css]: [comment that lingers for the entirety of the PR] The reason I named all of these fields `feed-feature` is because we call the feed, cards, and cluster all part of the "index" views. Renaming just the feed classes to index makes it more confusing in my opinion, since we have two other index views that are not going through this restyle.

File: knowledge_repo/app/templates/base.html [hunk 2]
```diff
@@ -26,6 +26,7 @@
         <script src="https://cdnjs.cloudflare.com/ajax/libs/html5shiv/3.6.2/html5shiv.js"></script>
         <script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/2.1.0/es5-shim.min.js"></script>
         <![endif]-->
+
```
Comment (NiharikaRay) [inline knowledge_repo/app/templates/base.html]: [nit] unrelated change?

File: knowledge_repo/app/templates/index-feed.html [hunk 3]
```diff
@@ -1,81 +1,72 @@
 {% extends "index-base.html" %}
 
-{% block inner_content %}
+{% block style_links %}
+{{ super() }}
+<link rel="stylesheet" href="{{ url_for('static', filename='css/pages/index.css')}}">
+{% endblock %}
+
+{% macro render_tag(tag, post_id) %}
```
Comment (NiharikaRay) [inline knowledge_repo/app/templates/index-feed.html]: [question] what was the reasoning for this macro? we only use it in this place, and it's made this diff a little harder to read (since the functionality hasn't changed, you've just made this a function).

File: knowledge_repo/app/templates/index-feed.html [hunk 4]
```diff
@@ -1,81 +1,72 @@
...
-          </p>
-        </div>
-        <div class="col-md-6">
-          <center><img src="{{ post['thumbnail'] if post['thumbnail'] else '/static/images/default_thumbnail.png' }}" style="max-width:100%;max-height:225px" /></center>
-        </div>
-      </div>
+{% block inner_content %}
+
+{% for post in posts %}
+<div class="row index-post" data-path='{{post.path|urlencode}}' data-url="{{url_for('posts.render', path=post.path)}}">
```
Comment (NiharikaRay) [inline knowledge_repo/app/templates/index-feed.html]: same comment about `index-post` vs `feed-post`

**Comment by NiharikaRay** [comment]
The hover over looks good, but the grey background on the hover panel is a little jarring. Could we change that?

File: knowledge_repo/app/static/css/custom.css (hunk 1 above)
Comment (matthewwardrop) [inline knowledge_repo/app/static/css/custom.css]: That's a fair point, and happy to comply. I was looking forward to the upcoming url revamp on the index routes whereby /feed becomes /index/feed (as documented earlier in an internal roadmap document). We also discussed briefly the fact that these alternative index views are potentially going to be deprecated by a more complete filtering interface, which would consolidate all of the index views into one /index route. The qualification to `index-post` instead of just `index` was supposed to allow other css classes to be present if required without clashing in the namespace. Perhaps I should have written all this in the PR description; but if you still convinced `feed-post` is a better name, I'll change them over. :).

As a side note, this won't make the diff look much different, since these css classes are defined in a separate css file (and only applied to the index/feed page to which they are relevant).

File: knowledge_repo/app/templates/base.html (hunk 2 above)
Comment (matthewwardrop) [inline knowledge_repo/app/templates/base.html]: Will remove. Made some changes here but ultimately reverted them.

File: knowledge_repo/app/templates/index-feed.html (hunk 3 above)
Comment (matthewwardrop) [inline knowledge_repo/app/templates/index-feed.html]: The macro just made it easier for me to read the html that renders the post. Apart from that, there was a mild interest in separating form from function. Most of the structure of the post rows are now semantic, whereas the html for the tags was a little more involves a little more structure. If you prefer it all inline, happy to rearrange it.

**Comment by matthewwardrop** [comment]
@NiharikaRay Will specify the background of the TLDR as an translucent color rather than a specified hex code, which should resolve this :).

File: knowledge_repo/app/static/css/custom.css (hunk 1 above)
Comment (NiharikaRay) [inline knowledge_repo/app/static/css/custom.css]: Got it. Given that this is pending based on future work happening, I think it probably makes sense to hold of on changing `feed-post` to `index-post`.

Fair point, i didn't realize you had re-arranged the css from one file to another.

File: knowledge_repo/app/templates/index-feed.html (hunk 3 above)
Comment (NiharikaRay) [inline knowledge_repo/app/templates/index-feed.html]: It's fine, I was just confused as to why since the rendering didn't change.

**Comment by NiharikaRay** [comment]
LGTM pending change from `index-post` to `feed-post`.

File: knowledge_repo/app/static/css/custom.css (hunk 1 above)
Comment (matthewwardrop) [inline knowledge_repo/app/static/css/custom.css]: Sounds good. Incoming :).
